packages/components/src/slot-fill/slot.tsx

```diff
@@ -13,20 +17,23 @@ import {
  * Internal dependencies
  */
 import SlotFillContext from './context';
+import type { BaseSlotComponentProps, SlotComponentProps } from './types';
 
 /**
  * Whether the argument is a function.
  *
  * @param {*} maybeFunc The argument to check.
  * @return {boolean} True if the argument is a function, false otherwise.
  */
-function isFunction( maybeFunc ) {
+function isFunction( maybeFunc: any ): maybeFunc is Function {
 	return typeof maybeFunc === 'function';
 }
 
-class SlotComponent extends Component {
-	constructor() {
-		super( ...arguments );
+class SlotComponent extends Component< BaseSlotComponentProps, {} > {
```

Comment:
The second `Component` generic argument is optional, no?

Reply:
`{}` is not necessary.